specs/vmclock.md

```diff
@@ -0,0 +1,325 @@
+---
+title: UAPI.13 VMClock
+category: Concepts
+layout: default
+version: 1.0
+SPDX-License-Identifier: CC-BY-4.0
+weight: 13
+aliases:
+- /UAPI.13
+- /13
+---
+
+# UAPI.13 VMClock: Efficient time synchronisation for virtual machines
+
+| Version | Changes |
+|---------|---------|
+| 1.0     | Initial Release |
+
+The requirements for accurate synchronisation of application clocks against
+real wallclock time are becoming ever more demanding. Increasingly cloud
+providers are exposing precision clock devices to virtual machines to allow the
+guest operating systems to synchronise their clocks.
+
+Time on modern systems is typically derived from a CPU-internal counter (*TSC,
+timebase, arch counter*) which runs at a nominally constant frequency of
+typically between 1GHz and 4GHz. In practice, the frequency of the underlying
+hardware counter will vary with environmental conditions, with a tolerance of
+the order of ±50PPM. It is this variance which must constantly be corrected by
```

Comment:
This might be totally clear to people in the time synchronization biz, but I was surprised that these PPM really do mean parts per million, so either a
```suggestion
the order of ±50PPM (parts per million). It is this variance which must constantly be corrected by
```
or a glossary at the bottom might be helpful.